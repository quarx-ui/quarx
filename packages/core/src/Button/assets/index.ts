import { ReactComponent as PaperClipIcon } from './paperClip24.svg';
import { ReactComponent as SmallPaperClipIcon } from './paperClip16.svg';
import { ReactComponent as ChevronDownIcon } from './chevronDown24.svg';

export { PaperClipIcon, SmallPaperClipIcon, ChevronDownIcon };
