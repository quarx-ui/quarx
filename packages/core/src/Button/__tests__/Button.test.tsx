import React, { FC, ReactNode } from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { expectPropsMapInClasses } from '@core/test-utils';
import { Button, IconButton, BaseButtonProps, ButtonStyleParams } from '..';
import { PaperClipIcon } from '../assets';

const checkPropsInClasses = (props: ButtonStyleParams) => {
    const {
        size = 'medium',
        borderRadius = 'smooth',
        color = 'brand',
        type = 'contained',
        loading = false,
        disabled = false,
    } = props;
    const buttonElement = screen.getByRole('button');
    const propsWithDefault = { size, borderRadius, color, type, loading, disabled };

    expectPropsMapInClasses(buttonElement)(propsWithDefault);
};

const checkProps = (Component: FC, renderedProps?: BaseButtonProps) => (checkedProps?: ButtonStyleParams) => () => {
    const { asFragment } = render(<Component {...renderedProps} {...checkedProps} />);

    checkPropsInClasses(checkedProps ?? {});
    expect(asFragment()).toMatchSnapshot();
};

const checkSnapshotsOf = (Component: FC, children: ReactNode) => {
    const checkButtonProps = checkProps(Component, { children });

    it('default', checkButtonProps());

    it('secondary', checkButtonProps({ color: 'secondary' }));

    it('xSmall', checkButtonProps({ size: 'xSmall' }));
    it('small', checkButtonProps({ size: 'small' }));
    it('large', checkButtonProps({ size: 'large' }));

    it('square', checkButtonProps({ borderRadius: 'square' }));
    it('rounded', checkButtonProps({ borderRadius: 'rounded' }));

    it('outlined', checkButtonProps({ type: 'outlined' }));
    it('text', checkButtonProps({ type: 'text' }));

    it('loading', checkButtonProps({ loading: true }));
    it('disabled', checkButtonProps({ disabled: true }));
};

const checkComponent = (Component: FC<BaseButtonProps>, children: ReactNode) => {
    it('loading should be in the document', () => {
        render(<Component loading>{children}</Component>);
        const button = screen.getByRole('button');
        const loader = button.children[1];

        expect(loader).toBeInTheDocument();
        expect(loader).toHaveClass(`Sx${Component.displayName}-loader`);
        expect(button).toHaveAttribute('disabled');
    });

    it('button should have attribute disabled', () => {
        const mockedOnClick = jest.fn();
        render(<Component disabled onClick={mockedOnClick}>{children}</Component>);
        const button = screen.getByRole('button');

        expect(button).toHaveAttribute('disabled');
        userEvent.click(button);
        expect(mockedOnClick).not.toBeCalled();
    });

    it('button should be hidden', () => {
        render(<Component hidden>{children}</Component>);
        expect(screen.queryByRole('button')).not.toBeInTheDocument();

        cleanup();

        render(<Component permissions={{ hidden: true }}>{children}</Component>);
        expect(screen.queryByRole('button')).not.toBeInTheDocument();
    });

    it('onClick should be called', () => {
        const mockedOnClick = jest.fn();
        render(<Component onClick={mockedOnClick}>{children}</Component>);
        const button = screen.getByRole('button');

        userEvent.click(button);
        expect(mockedOnClick).toBeCalledTimes(1);
    });
};

describe('Button snapshots', () => {
    checkSnapshotsOf(Button, 'Button');
});

describe('Button', () => {
    it('text should be in the document', () => {
        render(<Button>Button</Button>);

        expect(screen.queryByText('Button')).toBeInTheDocument();
        expect(screen.getByRole('button')).not.toHaveAttribute('disabled');
    });
    Button.displayName = 'Button';
    checkComponent(Button, 'Button');
});

describe('IconButton snapshots', () => {
    IconButton.displayName = 'IconButton';
    checkSnapshotsOf(IconButton, <PaperClipIcon />);
});

describe('IconButton', () => {
    it('svg should be in the document', () => {
        render(<IconButton><PaperClipIcon /></IconButton>);
        const svgElement = document.querySelector('svg');
        expect(svgElement).toBeInTheDocument();
    });
    checkComponent(IconButton, <PaperClipIcon />);
});
