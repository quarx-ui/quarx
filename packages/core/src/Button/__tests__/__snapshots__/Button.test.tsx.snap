// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshots default 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_smooth SxButton SxButton_size_medium css-3d2dfn-SxBaseButton-root-SxButton-root"
    type="button"
  >
    <div
      class="SxButton-content css-15h85ex-SxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots disabled 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_smooth SxBaseButton_disabled SxButton SxButton_size_medium css-j12tu0-SxBaseButton-root-SxButton-root"
    disabled=""
    type="button"
  >
    <div
      class="SxButton-content css-15h85ex-SxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots large 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_large SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_smooth SxButton SxButton_size_large css-12umxut-SxBaseButton-root-SxButton-root"
    type="button"
  >
    <div
      class="SxButton-content css-15h85ex-SxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots loading 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_smooth SxBaseButton_loading SxButton SxButton_size_medium SxButton_loading css-g2nmvz-SxBaseButton-root-SxButton-root"
    disabled=""
    type="button"
  >
    <div
      class="SxButton-content css-16bg0a3-SxButton-content"
    >
      Button
    </div>
    <div
      class="SxLoader SxLoader_size_base SxButton-loader css-1nxool5-SxLoader-root-SxButton-loader"
    >
      <div
        class="SxLoader-dot css-197lm1k-SxLoader-dot"
      />
      <div
        class="SxLoader-dot css-197lm1k-SxLoader-dot"
      />
      <div
        class="SxLoader-dot css-197lm1k-SxLoader-dot"
      />
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots outlined 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_brand SxBaseButton_type_outlined SxBaseButton_borderRadius_smooth SxButton SxButton_size_medium css-5wy14x-SxBaseButton-root-SxButton-root"
    type="button"
  >
    <div
      class="SxButton-content css-15h85ex-SxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots rounded 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_rounded SxButton SxButton_size_medium css-wx8y1m-SxBaseButton-root-SxButton-root"
    type="button"
  >
    <div
      class="SxButton-content css-15h85ex-SxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots secondary 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_secondary SxBaseButton_type_contained SxBaseButton_borderRadius_smooth SxButton SxButton_size_medium css-1o41i9p-SxBaseButton-root-SxButton-root"
    type="button"
  >
    <div
      class="SxButton-content css-15h85ex-SxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots small 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_small SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_smooth SxButton SxButton_size_small css-12os09q-SxBaseButton-root-SxButton-root"
    type="button"
  >
    <div
      class="SxButton-content css-15h85ex-SxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots square 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_square SxButton SxButton_size_medium css-1buc2yd-SxBaseButton-root-SxButton-root"
    type="button"
  >
    <div
      class="SxButton-content css-15h85ex-SxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots text 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_brand SxBaseButton_type_text SxBaseButton_borderRadius_smooth SxButton SxButton_size_medium css-1dmblyo-SxBaseButton-root-SxButton-root"
    type="button"
  >
    <div
      class="SxButton-content css-15h85ex-SxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots xSmall 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_xSmall SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_smooth SxButton SxButton_size_xSmall css-7g23n0-SxBaseButton-root-SxButton-root"
    type="button"
  >
    <div
      class="SxButton-content css-15h85ex-SxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton snapshots default 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_smooth SxIconButton SxIconButton_size_medium css-1mecpj1-SxBaseButton-root-SxIconButton-root"
    type="button"
  >
    <div
      class="SxIconButton-icon css-rgplqb-SxIconButton-icon"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.9755 13.1045C21.5163 10.5637 21.5149 6.44569 18.9746 3.9054C16.434 1.36487 12.315 1.36487 9.77451 3.9054L4.19078 9.48913C3.79993 9.87998 3.79993 10.5137 4.19078 10.9045C4.58163 11.2954 5.21532 11.2954 5.60617 10.9045L11.1899 5.32079C12.9487 3.56196 15.8003 3.56196 17.5592 5.32079C19.3182 7.07985 19.3187 9.9305 17.5601 11.6891L10.0262 19.2231C8.99154 20.2577 7.31534 20.2568 6.27848 19.22C5.24273 18.1842 5.24273 16.5049 6.27849 15.4692L13.5961 8.15158C13.9088 7.8389 14.4157 7.8389 14.7284 8.15158C15.0411 8.46426 15.0411 8.97122 14.7284 9.2839L9.35697 14.6553C8.96612 15.0462 8.96612 15.6799 9.35697 16.0707C9.74782 16.4616 10.3815 16.4616 10.7724 16.0707L16.1438 10.6993C17.2382 9.60491 17.2382 7.83057 16.1438 6.73619C15.0494 5.64181 13.2751 5.6418 12.1807 6.73619L4.86309 14.0538C3.04564 15.8712 3.04564 18.8179 4.86309 20.6354C6.67944 22.4517 9.62302 22.457 11.4416 20.6385L18.9755 13.1045Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton snapshots disabled 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_smooth SxBaseButton_disabled SxIconButton SxIconButton_size_medium css-ysshz5-SxBaseButton-root-SxIconButton-root"
    disabled=""
    type="button"
  >
    <div
      class="SxIconButton-icon css-rgplqb-SxIconButton-icon"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.9755 13.1045C21.5163 10.5637 21.5149 6.44569 18.9746 3.9054C16.434 1.36487 12.315 1.36487 9.77451 3.9054L4.19078 9.48913C3.79993 9.87998 3.79993 10.5137 4.19078 10.9045C4.58163 11.2954 5.21532 11.2954 5.60617 10.9045L11.1899 5.32079C12.9487 3.56196 15.8003 3.56196 17.5592 5.32079C19.3182 7.07985 19.3187 9.9305 17.5601 11.6891L10.0262 19.2231C8.99154 20.2577 7.31534 20.2568 6.27848 19.22C5.24273 18.1842 5.24273 16.5049 6.27849 15.4692L13.5961 8.15158C13.9088 7.8389 14.4157 7.8389 14.7284 8.15158C15.0411 8.46426 15.0411 8.97122 14.7284 9.2839L9.35697 14.6553C8.96612 15.0462 8.96612 15.6799 9.35697 16.0707C9.74782 16.4616 10.3815 16.4616 10.7724 16.0707L16.1438 10.6993C17.2382 9.60491 17.2382 7.83057 16.1438 6.73619C15.0494 5.64181 13.2751 5.6418 12.1807 6.73619L4.86309 14.0538C3.04564 15.8712 3.04564 18.8179 4.86309 20.6354C6.67944 22.4517 9.62302 22.457 11.4416 20.6385L18.9755 13.1045Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton snapshots large 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_large SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_smooth SxIconButton SxIconButton_size_large css-1ghzjzg-SxBaseButton-root-SxIconButton-root"
    type="button"
  >
    <div
      class="SxIconButton-icon css-rgplqb-SxIconButton-icon"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.9755 13.1045C21.5163 10.5637 21.5149 6.44569 18.9746 3.9054C16.434 1.36487 12.315 1.36487 9.77451 3.9054L4.19078 9.48913C3.79993 9.87998 3.79993 10.5137 4.19078 10.9045C4.58163 11.2954 5.21532 11.2954 5.60617 10.9045L11.1899 5.32079C12.9487 3.56196 15.8003 3.56196 17.5592 5.32079C19.3182 7.07985 19.3187 9.9305 17.5601 11.6891L10.0262 19.2231C8.99154 20.2577 7.31534 20.2568 6.27848 19.22C5.24273 18.1842 5.24273 16.5049 6.27849 15.4692L13.5961 8.15158C13.9088 7.8389 14.4157 7.8389 14.7284 8.15158C15.0411 8.46426 15.0411 8.97122 14.7284 9.2839L9.35697 14.6553C8.96612 15.0462 8.96612 15.6799 9.35697 16.0707C9.74782 16.4616 10.3815 16.4616 10.7724 16.0707L16.1438 10.6993C17.2382 9.60491 17.2382 7.83057 16.1438 6.73619C15.0494 5.64181 13.2751 5.6418 12.1807 6.73619L4.86309 14.0538C3.04564 15.8712 3.04564 18.8179 4.86309 20.6354C6.67944 22.4517 9.62302 22.457 11.4416 20.6385L18.9755 13.1045Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton snapshots loading 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_smooth SxBaseButton_loading SxIconButton SxIconButton_size_medium SxIconButton_loading css-1k77p9v-SxBaseButton-root-SxIconButton-root"
    disabled=""
    type="button"
  >
    <div
      class="SxIconButton-icon css-qudlwq-SxIconButton-icon"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.9755 13.1045C21.5163 10.5637 21.5149 6.44569 18.9746 3.9054C16.434 1.36487 12.315 1.36487 9.77451 3.9054L4.19078 9.48913C3.79993 9.87998 3.79993 10.5137 4.19078 10.9045C4.58163 11.2954 5.21532 11.2954 5.60617 10.9045L11.1899 5.32079C12.9487 3.56196 15.8003 3.56196 17.5592 5.32079C19.3182 7.07985 19.3187 9.9305 17.5601 11.6891L10.0262 19.2231C8.99154 20.2577 7.31534 20.2568 6.27848 19.22C5.24273 18.1842 5.24273 16.5049 6.27849 15.4692L13.5961 8.15158C13.9088 7.8389 14.4157 7.8389 14.7284 8.15158C15.0411 8.46426 15.0411 8.97122 14.7284 9.2839L9.35697 14.6553C8.96612 15.0462 8.96612 15.6799 9.35697 16.0707C9.74782 16.4616 10.3815 16.4616 10.7724 16.0707L16.1438 10.6993C17.2382 9.60491 17.2382 7.83057 16.1438 6.73619C15.0494 5.64181 13.2751 5.6418 12.1807 6.73619L4.86309 14.0538C3.04564 15.8712 3.04564 18.8179 4.86309 20.6354C6.67944 22.4517 9.62302 22.457 11.4416 20.6385L18.9755 13.1045Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="SxLoader SxLoader_size_base SxIconButton-loader css-1yd4sv8-SxLoader-root-SxIconButton-loader"
    >
      <div
        class="SxLoader-dot css-197lm1k-SxLoader-dot"
      />
      <div
        class="SxLoader-dot css-197lm1k-SxLoader-dot"
      />
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton snapshots outlined 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_brand SxBaseButton_type_outlined SxBaseButton_borderRadius_smooth SxIconButton SxIconButton_size_medium css-y5hfjn-SxBaseButton-root-SxIconButton-root"
    type="button"
  >
    <div
      class="SxIconButton-icon css-rgplqb-SxIconButton-icon"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.9755 13.1045C21.5163 10.5637 21.5149 6.44569 18.9746 3.9054C16.434 1.36487 12.315 1.36487 9.77451 3.9054L4.19078 9.48913C3.79993 9.87998 3.79993 10.5137 4.19078 10.9045C4.58163 11.2954 5.21532 11.2954 5.60617 10.9045L11.1899 5.32079C12.9487 3.56196 15.8003 3.56196 17.5592 5.32079C19.3182 7.07985 19.3187 9.9305 17.5601 11.6891L10.0262 19.2231C8.99154 20.2577 7.31534 20.2568 6.27848 19.22C5.24273 18.1842 5.24273 16.5049 6.27849 15.4692L13.5961 8.15158C13.9088 7.8389 14.4157 7.8389 14.7284 8.15158C15.0411 8.46426 15.0411 8.97122 14.7284 9.2839L9.35697 14.6553C8.96612 15.0462 8.96612 15.6799 9.35697 16.0707C9.74782 16.4616 10.3815 16.4616 10.7724 16.0707L16.1438 10.6993C17.2382 9.60491 17.2382 7.83057 16.1438 6.73619C15.0494 5.64181 13.2751 5.6418 12.1807 6.73619L4.86309 14.0538C3.04564 15.8712 3.04564 18.8179 4.86309 20.6354C6.67944 22.4517 9.62302 22.457 11.4416 20.6385L18.9755 13.1045Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton snapshots rounded 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_rounded SxIconButton SxIconButton_size_medium css-1lpimna-SxBaseButton-root-SxIconButton-root"
    type="button"
  >
    <div
      class="SxIconButton-icon css-rgplqb-SxIconButton-icon"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.9755 13.1045C21.5163 10.5637 21.5149 6.44569 18.9746 3.9054C16.434 1.36487 12.315 1.36487 9.77451 3.9054L4.19078 9.48913C3.79993 9.87998 3.79993 10.5137 4.19078 10.9045C4.58163 11.2954 5.21532 11.2954 5.60617 10.9045L11.1899 5.32079C12.9487 3.56196 15.8003 3.56196 17.5592 5.32079C19.3182 7.07985 19.3187 9.9305 17.5601 11.6891L10.0262 19.2231C8.99154 20.2577 7.31534 20.2568 6.27848 19.22C5.24273 18.1842 5.24273 16.5049 6.27849 15.4692L13.5961 8.15158C13.9088 7.8389 14.4157 7.8389 14.7284 8.15158C15.0411 8.46426 15.0411 8.97122 14.7284 9.2839L9.35697 14.6553C8.96612 15.0462 8.96612 15.6799 9.35697 16.0707C9.74782 16.4616 10.3815 16.4616 10.7724 16.0707L16.1438 10.6993C17.2382 9.60491 17.2382 7.83057 16.1438 6.73619C15.0494 5.64181 13.2751 5.6418 12.1807 6.73619L4.86309 14.0538C3.04564 15.8712 3.04564 18.8179 4.86309 20.6354C6.67944 22.4517 9.62302 22.457 11.4416 20.6385L18.9755 13.1045Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton snapshots secondary 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_secondary SxBaseButton_type_contained SxBaseButton_borderRadius_smooth SxIconButton SxIconButton_size_medium css-u3lvmm-SxBaseButton-root-SxIconButton-root"
    type="button"
  >
    <div
      class="SxIconButton-icon css-rgplqb-SxIconButton-icon"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.9755 13.1045C21.5163 10.5637 21.5149 6.44569 18.9746 3.9054C16.434 1.36487 12.315 1.36487 9.77451 3.9054L4.19078 9.48913C3.79993 9.87998 3.79993 10.5137 4.19078 10.9045C4.58163 11.2954 5.21532 11.2954 5.60617 10.9045L11.1899 5.32079C12.9487 3.56196 15.8003 3.56196 17.5592 5.32079C19.3182 7.07985 19.3187 9.9305 17.5601 11.6891L10.0262 19.2231C8.99154 20.2577 7.31534 20.2568 6.27848 19.22C5.24273 18.1842 5.24273 16.5049 6.27849 15.4692L13.5961 8.15158C13.9088 7.8389 14.4157 7.8389 14.7284 8.15158C15.0411 8.46426 15.0411 8.97122 14.7284 9.2839L9.35697 14.6553C8.96612 15.0462 8.96612 15.6799 9.35697 16.0707C9.74782 16.4616 10.3815 16.4616 10.7724 16.0707L16.1438 10.6993C17.2382 9.60491 17.2382 7.83057 16.1438 6.73619C15.0494 5.64181 13.2751 5.6418 12.1807 6.73619L4.86309 14.0538C3.04564 15.8712 3.04564 18.8179 4.86309 20.6354C6.67944 22.4517 9.62302 22.457 11.4416 20.6385L18.9755 13.1045Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton snapshots small 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_small SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_smooth SxIconButton SxIconButton_size_small css-153m9cl-SxBaseButton-root-SxIconButton-root"
    type="button"
  >
    <div
      class="SxIconButton-icon css-rgplqb-SxIconButton-icon"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.9755 13.1045C21.5163 10.5637 21.5149 6.44569 18.9746 3.9054C16.434 1.36487 12.315 1.36487 9.77451 3.9054L4.19078 9.48913C3.79993 9.87998 3.79993 10.5137 4.19078 10.9045C4.58163 11.2954 5.21532 11.2954 5.60617 10.9045L11.1899 5.32079C12.9487 3.56196 15.8003 3.56196 17.5592 5.32079C19.3182 7.07985 19.3187 9.9305 17.5601 11.6891L10.0262 19.2231C8.99154 20.2577 7.31534 20.2568 6.27848 19.22C5.24273 18.1842 5.24273 16.5049 6.27849 15.4692L13.5961 8.15158C13.9088 7.8389 14.4157 7.8389 14.7284 8.15158C15.0411 8.46426 15.0411 8.97122 14.7284 9.2839L9.35697 14.6553C8.96612 15.0462 8.96612 15.6799 9.35697 16.0707C9.74782 16.4616 10.3815 16.4616 10.7724 16.0707L16.1438 10.6993C17.2382 9.60491 17.2382 7.83057 16.1438 6.73619C15.0494 5.64181 13.2751 5.6418 12.1807 6.73619L4.86309 14.0538C3.04564 15.8712 3.04564 18.8179 4.86309 20.6354C6.67944 22.4517 9.62302 22.457 11.4416 20.6385L18.9755 13.1045Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton snapshots square 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_square SxIconButton SxIconButton_size_medium css-xo0ul8-SxBaseButton-root-SxIconButton-root"
    type="button"
  >
    <div
      class="SxIconButton-icon css-rgplqb-SxIconButton-icon"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.9755 13.1045C21.5163 10.5637 21.5149 6.44569 18.9746 3.9054C16.434 1.36487 12.315 1.36487 9.77451 3.9054L4.19078 9.48913C3.79993 9.87998 3.79993 10.5137 4.19078 10.9045C4.58163 11.2954 5.21532 11.2954 5.60617 10.9045L11.1899 5.32079C12.9487 3.56196 15.8003 3.56196 17.5592 5.32079C19.3182 7.07985 19.3187 9.9305 17.5601 11.6891L10.0262 19.2231C8.99154 20.2577 7.31534 20.2568 6.27848 19.22C5.24273 18.1842 5.24273 16.5049 6.27849 15.4692L13.5961 8.15158C13.9088 7.8389 14.4157 7.8389 14.7284 8.15158C15.0411 8.46426 15.0411 8.97122 14.7284 9.2839L9.35697 14.6553C8.96612 15.0462 8.96612 15.6799 9.35697 16.0707C9.74782 16.4616 10.3815 16.4616 10.7724 16.0707L16.1438 10.6993C17.2382 9.60491 17.2382 7.83057 16.1438 6.73619C15.0494 5.64181 13.2751 5.6418 12.1807 6.73619L4.86309 14.0538C3.04564 15.8712 3.04564 18.8179 4.86309 20.6354C6.67944 22.4517 9.62302 22.457 11.4416 20.6385L18.9755 13.1045Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton snapshots text 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_medium SxBaseButton_color_brand SxBaseButton_type_text SxBaseButton_borderRadius_smooth SxIconButton SxIconButton_size_medium css-2gi83a-SxBaseButton-root-SxIconButton-root"
    type="button"
  >
    <div
      class="SxIconButton-icon css-rgplqb-SxIconButton-icon"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.9755 13.1045C21.5163 10.5637 21.5149 6.44569 18.9746 3.9054C16.434 1.36487 12.315 1.36487 9.77451 3.9054L4.19078 9.48913C3.79993 9.87998 3.79993 10.5137 4.19078 10.9045C4.58163 11.2954 5.21532 11.2954 5.60617 10.9045L11.1899 5.32079C12.9487 3.56196 15.8003 3.56196 17.5592 5.32079C19.3182 7.07985 19.3187 9.9305 17.5601 11.6891L10.0262 19.2231C8.99154 20.2577 7.31534 20.2568 6.27848 19.22C5.24273 18.1842 5.24273 16.5049 6.27849 15.4692L13.5961 8.15158C13.9088 7.8389 14.4157 7.8389 14.7284 8.15158C15.0411 8.46426 15.0411 8.97122 14.7284 9.2839L9.35697 14.6553C8.96612 15.0462 8.96612 15.6799 9.35697 16.0707C9.74782 16.4616 10.3815 16.4616 10.7724 16.0707L16.1438 10.6993C17.2382 9.60491 17.2382 7.83057 16.1438 6.73619C15.0494 5.64181 13.2751 5.6418 12.1807 6.73619L4.86309 14.0538C3.04564 15.8712 3.04564 18.8179 4.86309 20.6354C6.67944 22.4517 9.62302 22.457 11.4416 20.6385L18.9755 13.1045Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton snapshots xSmall 1`] = `
<DocumentFragment>
  <button
    class="SxBaseButton SxBaseButton_size_xSmall SxBaseButton_color_brand SxBaseButton_type_contained SxBaseButton_borderRadius_smooth SxIconButton SxIconButton_size_xSmall css-fn6ci0-SxBaseButton-root-SxIconButton-root"
    type="button"
  >
    <div
      class="SxIconButton-icon css-rgplqb-SxIconButton-icon"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.9755 13.1045C21.5163 10.5637 21.5149 6.44569 18.9746 3.9054C16.434 1.36487 12.315 1.36487 9.77451 3.9054L4.19078 9.48913C3.79993 9.87998 3.79993 10.5137 4.19078 10.9045C4.58163 11.2954 5.21532 11.2954 5.60617 10.9045L11.1899 5.32079C12.9487 3.56196 15.8003 3.56196 17.5592 5.32079C19.3182 7.07985 19.3187 9.9305 17.5601 11.6891L10.0262 19.2231C8.99154 20.2577 7.31534 20.2568 6.27848 19.22C5.24273 18.1842 5.24273 16.5049 6.27849 15.4692L13.5961 8.15158C13.9088 7.8389 14.4157 7.8389 14.7284 8.15158C15.0411 8.46426 15.0411 8.97122 14.7284 9.2839L9.35697 14.6553C8.96612 15.0462 8.96612 15.6799 9.35697 16.0707C9.74782 16.4616 10.3815 16.4616 10.7724 16.0707L16.1438 10.6993C17.2382 9.60491 17.2382 7.83057 16.1438 6.73619C15.0494 5.64181 13.2751 5.6418 12.1807 6.73619L4.86309 14.0538C3.04564 15.8712 3.04564 18.8179 4.86309 20.6354C6.67944 22.4517 9.62302 22.457 11.4416 20.6385L18.9755 13.1045Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;
